### ISSUE TITLE
Crash on Android when on x86_64 (vs x86)

### ISSUE THREAD

**Issue by tenzap** [libexif/libexif#96]
Not sure this is a bug in libexif/exif but maybe you have an idea as to where to look at since you maintain the C code that I have some difficulty to read.

You will find some information on the issue I'm facing here:

https://stackoverflow.com/questions/75118405/crash-on-vsprintf-starting-from-android12-api-31

**Comment by dfandrich** [comment]
What happens when you run the same input file through the command-line exif
tool? There shouldn't be any effective difference between the vfprintf in the
original code and the vsprintf in your version. Are you sure it isn't a bug
in this version of Bionic?

**Comment by tenzap** [comment]
Thanks for helping.

I managed to build exif for android (it required to build also `popt` which BTW only works starting from API28 since it requires 'glob').

A simple `exif pic.jpg` reports the warning correctly while through my app it crashes instead of reporting the error as it did usually in the past.

In the meantime since API30 images have x86 arch, and API31 images are x86_64 I just tried API30 with a x86_64. For now:
- API 30 x86 → no crash in app
- API 30 x86_64 → crash in app (exif command line in shell reports the expected warning)
- API 31 x86_64 → crash in app (exif command line in shell reports the expected warning)

In the meantime I found that the args passed to vsprintf  don't have the expected values. So the problem is likely not vsprintf. Concerning bionic, how to say... Still investigating. Myabe a 32/64bit issue?
